client/src/__tests__/search.test.js

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react'
+import {render, screen, waitFor} from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import "mutationobserver-shim"
+import Search from '../components/Search'
+
+jest.setTimeout(30000);
```

Comment:
Why do we need this?

Reply:
Hi @laredotornado  - Are you asking about the mutationobserver-shim? I added it for a few reasons:
* When I run the test without it, I get the error " TypeError: MutationObserver is not a constructor" on the line "await waitFor(() => screen.findByText(/Loading/i))."
* I found mutationobserver-shim as a solution on this StackOverflow post: https://stackoverflow.com/questions/64188000/mutation-observer-is-not-a-constructor-jest.
* Some other StackOverflow posts have suggested adding a devDependency on "jest-environment-jsdom-sixteen", but I didn't try that once I found that mutationobserver-shim worked. I can try jest-environment-jsdom-sixteen if you prefer. (StackOverflow link: https://stackoverflow.com/questions/61036156/react-typescript-testing-typeerror-mutationobserver-is-not-a-constructor)